### ISSUE TITLE
Segmentation fault in LibraryImportGenerator.Unit.Tests on coreclr/linux/arm64/debug

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#70112]
Segfault on an unrelated PR - https://github.com/dotnet/runtime/pull/66226, [build](https://dev.azure.com/dnceng/public/_build/results?buildId=1799145&view=logs&jobId=e4f18bbc-2f57-55e3-8831-999b74ca15c3&j=e4f18bbc-2f57-55e3-8831-999b74ca15c3&t=7dbfac67-24c2-5451-1abf-2cb25235c000), and [log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-66226-merge-99212800b2394b7f9e/LibraryImportGenerator.Unit.Tests/1/console.3a0a4793.log?helixlogtype=result):

```
===========================================================================================================
/root/helix/work/workitem/e /root/helix/work/workitem/e
  Discovering: LibraryImportGenerator.Unit.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  LibraryImportGenerator.Unit.Tests (found 122 of 127 test cases)
  Starting:    LibraryImportGenerator.Unit.Tests (parallel test collections = on, max threads = 4)
    LibraryImportGenerator.UnitTests.Compiles.ValidateSnippetsWithMarshalType [SKIP]
      No current scenarios to test.
./RunTests.sh: line 168:    24 Segmentation fault      (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig LibraryImportGenerator.Unit.Tests.runtimeconfig.json --depsfile LibraryImportGenerator.Unit.Tests.deps.json xunit.console.dll LibraryImportGenerator.Unit.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing $RSP_FILE
/root/helix/work/workitem/e
----- end Wed Jun 1 01:05:25 UTC 2022 ----- exit code 139 ----------------------------------------------------------
exit code 139 means SIGSEGV Illegal memory access. Deref invalid pointer, overrunning buffer, stack overflow etc. Core dumped.
```

Maybe similar to https://github.com/dotnet/runtime/issues/68443

cc @jkotas @danmoseley

**Comment by jakobbotsch** [comment]
This could be #69657 that was fixed in #70053 too.

**Comment by jkotas** [comment]
I am not able to get a good unmanaged stacktrace from the crashdump (the symbols are mismatched for some reason).

> This could be https://github.com/dotnet/runtime/issues/69657

Yes, that's likely. Let's resolve it with this assumption.
